plugins/inputs/opcua/opcua_client.go

```diff
@@ -593,6 +637,7 @@ func init() {
 			Certificate:    "/etc/telegraf/cert.pem",
 			PrivateKey:     "/etc/telegraf/key.pem",
 			AuthMethod:     "Anonymous",
+			codes:          []ua.StatusCode{ua.StatusOK},
```

Comment:
Would it make sense to force addition of the OK status code so it is always accepted (i.e. not overwritable by the user)?

Reply:
I can't imagine a case where the user would not want StatusOK to be a valid status so this makes sense to me.

Reply:
@R290 I don't see the change in the code, so I unresolved it.  The idea here is to _always_ add the `StatusOK` to the `codes`. This has to be done in `Init()` as otherwise the user might overwrite the setting.
Do you have any use-cases where `StatusOK` is _not_ a valid return code?

Reply:
I think my last comment here was ambiguous. Since I can't imagine a case where StatusOK is not a valid code, I don't think it should be required in the new codes setting. I like srebhan's idea of having the code always accept StatusOK even if the user doesn't include it in the codes setting.

@R290 Maybe we should rename the setting to be additional_valid_status_codes so people know they're in addition to StatusOK.

Reply:
@srebhan I thought I had this covered by initializing `o.codes` in the lower case `init()` function and no longer clearing `o.codes`: https://github.com/influxdata/telegraf/pull/10384/commits/cc971097c71fad5eb0add3db9c22cfda7367ff54#diff-50ccc8f6ff9c3271b29a8f73ed1d04c026262c0f79849b176c605eda10eb575cL502. Additional valid status can only be appended and `o.codes` should always contain `ua.StatusOK` this way? I agree that there is no reason for not including `ua.StatusOK`. 

@reimda see https://github.com/influxdata/telegraf/pull/10384/commits/5691c5fb5ab2ff9ddc14fbae679f165b4905c4fa for the renaming.

Reply:
@R290 ok now I see. You are absolutely correct. This is exactly what we want.
```